editoast/src/client/roles.rs

```diff
@@ -0,0 +1,136 @@
+use std::collections::HashSet;
+
+use anyhow::{anyhow, bail};
+use clap::{Args, Subcommand};
+use editoast_authz::{
+    authorizer::{StorageDriver, UserInfo},
+    BuiltinRole,
+};
+use editoast_models::DbConnection;
+use itertools::Itertools as _;
+use strum::IntoEnumIterator;
+use tracing::info;
+
+use crate::models::auth::PgAuthDriver;
+
+#[derive(Debug, Subcommand)]
+pub enum RolesCommand {
+    /// Lists the builtin roles supported by editoast
+    ListRoles,
+    /// Lists the roles assigned to a subject
+    List(ListArgs),
+    /// Grants builtin roles to a subject
+    Add(AddArgs),
+    /// Revokes builtin roles from a subject
+    Remove(RemoveArgs),
+}
+
+#[derive(Debug, Args)]
+pub struct ListArgs {
+    /// A subject ID or user identity
+    subject: String,
+}
+
+#[derive(Debug, Args)]
+pub struct AddArgs {
+    /// A subject ID or user identity
+    subject: String,
+    /// A non-empty list of builtin roles
+    roles: Vec<String>,
+}
+
+#[derive(Debug, Args)]
+pub struct RemoveArgs {
+    /// A subject ID or user identity
+    subject: String,
+    /// A non-empty list of builtin roles
+    roles: Vec<String>,
+}
+
+pub fn list_roles() {
+    BuiltinRole::iter().for_each(|role| println!("{role}"));
+}
+
+async fn parse_and_fetch_subject_id(
+    subject: &String,
+    driver: &PgAuthDriver<BuiltinRole>,
+) -> anyhow::Result<i64> {
```

Comment:
We could return more information to have more user friendly prompt:

```diff
$ editoast roles list 2
- Subject 2 has no roles assigned
+ Subject flomonster (2) has no roles assigned
```

```suggestion
) -> anyhow::Result<UserInfo> {
```